src/filesystem/FileSystem.ts

```diff
@@ -859,48 +858,53 @@ define(function (require, exports, module) {
      *      completed. If the watch fails, the function will have a non-null FileSystemError
      *      string parametr.
      */
-    FileSystem.prototype.watch = function (entry, filter, filterGlobs, callback) {
+    public watch(entry, filter, filterGlobs, callback) {
         // make filterGlobs an optional argument to stay backwards compatible
         if (typeof callback === "undefined" && typeof filterGlobs === "function") {
             callback = filterGlobs;
             filterGlobs = null;
         }
 
-        var fullPath = entry.fullPath;
+        const fullPath = entry.fullPath;
 
-        callback = callback || function () {};
+        callback = callback || function () { /* Do nothing */ };
 
-        var watchingParentRoot = this._findWatchedRootForPath(fullPath);
+        const watchingParentRoot = this._findWatchedRootForPath(fullPath);
         if (watchingParentRoot &&
                 (watchingParentRoot.status === WatchedRoot.STARTING ||
-                 watchingParentRoot.status === WatchedRoot.ACTIVE)) {
+                watchingParentRoot.status === WatchedRoot.ACTIVE)) {
             callback("A parent of this root is already watched");
             return;
         }
 
-        var watchingChildRoot = Object.keys(this._watchedRoots).some(function (path) {
-            var watchedRoot = this._watchedRoots[path],
-                watchedPath = watchedRoot.entry.fullPath;
-
-            return watchedPath.indexOf(fullPath) === 0;
-        }, this);
+        let watchingChildRoot: WatchedRoot | null = null;
+        for (const path in this._watchedRoots) {
+            if (this._watchedRoots.hasOwnProperty(path)) {
+                const watchedRoot = this._watchedRoots[path];
+                const watchedPath = watchedRoot.entry.fullPath;
+                if (watchedPath.indexOf(fullPath) === 0) {
+                    watchingChildRoot = watchedRoot;
+                    break;
+                }
+            }
+        }
```

Comment:
This is a change in behaviour: before the loop always returned a boolean, now it returns what was supposed to return.